supplementary/alica_templates/Expr/src/ActionServerExample2379894799421542548.cpp

```diff
@@ -0,0 +1,137 @@
+    _actionServer->start();
+    /*PROTECTED REGION END*/
+}
+ActionServerExample2379894799421542548::~ActionServerExample2379894799421542548()
+{
+    /*PROTECTED REGION ID(dcon2379894799421542548) ENABLED START*/
+    // Add additional options here
+    /*PROTECTED REGION END*/
+}
+
+/**
+ * Task: DefaultTask  -> EntryPoint-ID: 1647616282106629095
+ */
+std::shared_ptr<UtilityFunction> UtilityFunction2379894799421542548::getUtilityFunction(Plan* plan)
+{
+    /*PROTECTED REGION ID(2379894799421542548) ENABLED START*/
+    std::shared_ptr<UtilityFunction> defaultFunction = std::make_shared<DefaultUtilityFunction>(plan);
+    return defaultFunction;
+    /*PROTECTED REGION END*/
+}
+
+/**
+ * Transition: 430744406068167347 (430744406068167347)
+ *   - Comment:
+ *   - Source2Dest: WaitForGoal --> ExecuteGoal
+ *
+ * Precondition: 1886820548377048134 (1886820548377048134)
+ *   - Enabled: true
+ *   - PluginName: DefaultPlugin
+ *   - ConditionString:
+ *   - Variables:
+ *   - Quantifiers:
+ *
+ * Abstract Plans in WaitForGoal:
+ */
+bool PreCondition1886820548377048134::evaluate(std::shared_ptr<RunningPlan> rp, const IAlicaWorldModel* wm)
+{
+    /*PROTECTED REGION ID(430744406068167347) ENABLED START*/
+    LockedBlackboardRO bb = LockedBlackboardRO(*(rp->getBasicPlan()->getBlackboard()));
+    return bb.get<std::optional<int32_t>>("goal").has_value();
+    /*PROTECTED REGION END*/
+}
+
+/**
+ * Transition: 1354699620997961969 (1354699620997961969)
+ *   - Comment:
+ *   - Source2Dest: ExecuteGoal --> WaitForGoal
+ *
+ * Precondition: 587249152722263568 (587249152722263568)
+ *   - Enabled: true
+ *   - PluginName: DefaultPlugin
+ *   - ConditionString:
+ *   - Variables:
+ *   - Quantifiers:
+ *
+ * Abstract Plans in ExecuteGoal:
+ *   - DummyImplementation (4126421719858579722)
+ */
+bool PreCondition587249152722263568::evaluate(std::shared_ptr<RunningPlan> rp, const IAlicaWorldModel* wm)
+{
+    /*PROTECTED REGION ID(1354699620997961969) ENABLED START*/
+    LockedBlackboardRO bb = LockedBlackboardRO(*(rp->getBasicPlan()->getBlackboard()));
+    return rp->isAnyChildTaskSuccessful() || rp->isAnyChildStatus(PlanStatus::Failed) || bb.get<std::optional<bool>>("cancel").value_or(false);
+    /*PROTECTED REGION END*/
+}
+
+/*PROTECTED REGION ID(methods2379894799421542548) ENABLED START*/
+// Add additional options here
+void ActionServerExample2379894799421542548::onInit()
+{
+    LockedBlackboardRW bb = LockedBlackboardRW(*getBlackboard());
+    bb.registerValue("result", std::optional<int32_t>());
+    bb.registerValue("feedback", std::optional<int32_t>());
+    bb.registerValue("goal", std::optional<int32_t>());
+    bb.registerValue("cancel", std::optional<bool>());
+}
+
+void ActionServerExample2379894799421542548::run(void* msg)
+{
+    LockedBlackboardRW bb = LockedBlackboardRW(*getBlackboard());
+    if (_actionServer->isNewGoalAvailable()) {
+        bb.get<std::optional<int32_t>>("cancel") = false;
```

Comment:
Need not explicitly set this since you are checking with value_or anyways in condition. "cancel" could be just empty and be set to "true" when needed